go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/juicedata/juicefs
 
-go 1.18
```

Comment:
I tested without upgrading and it's ok here.

Reply:
Yeah, the version number here serves two purposes
* Other module users: Minimal required Go version number to build - from this perspective it requires upgrade
* Go compiler: Defines default behaviour in case it changes between versions - from this perspective upgrade is not required, but with enough test coverage it can be safely done in this case.

Reply:
If we wait for #4334 this pr to merge, we shouldn't have this problem anymore

Reply:
Either one is merged first, it is a minor and easy conflict to resolve.

Reply:
Reverted to 1.18 in 621e9ef80b311beb2d8bbadc72a9c0c790a450c9.